### PR TITLE
doc: add list of valid components to the development guide

### PR DESCRIPTION
**Pull request by nixpanic**
# Describe what this PR does #

The commitlint CI job uses the configuration from .commitlintrc.yaml
which contains the different components that Ceph-CSI uses. A short
description of each component has been added, so that contributors
understand what component to mention in the prefix of the subject in
commit messages.

## Related issues ##

See-also: #1058

### CODE REVIEW

**Comment by nixpanic** [comment]
I don't know what is not semantic about the commit message (or PR), @humblec do you know?

File: docs/development-guide.md [hunk 1]
```diff
@@ -122,13 +122,30 @@ Signed-off-by: Random J Developer <random@developer.example.org>
...
+<component>: <subject of the change>
 <BLANK LINE>
-<why this change was made>
+<paragraph(s) with reason/description>
 <BLANK LINE>
-<footer>
+<signed-off-by>
 ```
 
+The `component` in the subject of the commit message can be one of teh following:
```
Comment (Madhu-1) [inline docs/development-guide.md:132]: the following

File: docs/development-guide.md [hunk 2]
```diff
@@ -122,13 +122,30 @@ Signed-off-by: Random J Developer <random@developer.example.org>
...
+
+* `cephfs`: bugs or enhancements related to CephFS
+* `rbd`: bugs or enhancements related to RBD images
```
Comment (Madhu-1) [inline docs/development-guide.md:135]: Related to RBD not RBD images

File: docs/development-guide.md [hunk 3]
```diff
@@ -122,13 +122,30 @@ Signed-off-by: Random J Developer <random@developer.example.org>
...
+* `doc`: documentation updates
+* `util`: utilities shared betweed componentsm use `cephfs` or `rbd` if the
```
Comment (Madhu-1) [inline docs/development-guide.md:137]: Between Components

**Comment by nixpanic** [comment]
> Need to add a component for kubernetes template.

Lets do that afterwards. There needs to be a `rebase` one as well, I think.

File: docs/development-guide.md [hunk 4]
```diff
@@ -122,13 +122,30 @@ Signed-off-by: Random J Developer <random@developer.example.org>
...
+   change is only relevant for one of the type of storage
+* `journal`: any of the journalling functionalities
+* `helm`: deployment changes for the Helm charts
+* `build`: anything related to building Ceph-CSI, the executable or container
+   images
+* `ci`: changes related to the Continuous Integration, or testing
+* `e2e`: end-to-end testing updates
+* `cleanup`: general maintenance and cleanup changes
+* `revert`: undo a commit that was merged by mistake, use of one of the other
+   components is in most cases recommended
```
Comment (Madhu-1) [inline docs/development-guide.md:147]:
```suggestion
* `revert`: undo a commit that was merged by mistake

use of one of the other components is in most cases recommended
```

Comment (nixpanic) [reply docs/development-guide.md:147]: I am not sure what you mean here. There are no 'other components' that are not listed here.

Comment (Madhu-1) [reply docs/development-guide.md:147]: nevermind i was thinking something else

File: docs/development-guide.md [hunk 5]
```diff
@@ -122,13 +122,30 @@ Signed-off-by: Random J Developer <random@developer.example.org>
...
+* `util`: utilities shared between components use `cephfs` or `rbd` if the
+   change is only relevant for one of the type of storage
+* `journal`: any of the journalling functionalities
+* `helm`: deployment changes for the Helm charts
+* `build`: anything related to building Ceph-CSI, the executable or container
+   images
+* `ci`: changes related to the Continuous Integration, or testing
+* `e2e`: end-to-end testing updates
+* `cleanup`: general maintenance and cleanup changes
+* `revert`: undo a commit that was merged by mistake, use of one of the other
```
Comment (Madhu-1) [inline docs/development-guide.md:146]: can we need to split this line, `use of one of the other components is in most cases recommended` can be a separate line?

**Comment by Madhu-1** [comment]
`Semantic Pull Request` is failing. any idea?

File: docs/development-guide.md (hunk 5 above)
Comment (nixpanic) [reply docs/development-guide.md:146]: I really intended to have a note about `revert`, it should almost never be used. Adding the `use of one of the other components is in most cases recommended` on its own line does not make much sense to me.

**Comment by nixpanic** [comment]
> `Semantic Pull Request` is failing. any idea?

No idea what so ever... the `.github/semantic.yaml` does have `doc` as prefix for a commit message listed :open_mouth:

**Comment by Madhu-1** [comment]
> > `Semantic Pull Request` is failing. any idea?
> 
> No idea what so ever... the `.github/semantic.yaml` does have `doc` as prefix for a commit message listed

let's find out why it's failing and what is the format it's expecting(we need to document the same). if this is more problematic we can remove the `Sematic Pull  Request` bot from the repo.

**Comment by nixpanic** [comment]
> till we figure out go we need sematic PR check or not

It is the replacement for the commitlint bot that is not maintained anymore. Maybe we should just move to [Prow](https://github.com/kubernetes/test-infra/blob/master/prow/README.md) and deploy it on the CentOS CI OpenShift project we have.

**Comment by nixpanic** [comment]
The PR text has now has been updated to reflect the changes for #1058 and the assumption of #1057

File: docs/development-guide.md [hunk 6]
```diff
@@ -122,13 +122,30 @@ Signed-off-by: Random J Developer <random@developer.example.org>
...
+The `component` in the subject of the commit message can be one of the following:
+
+* `cephfs`: bugs or enhancements related to CephFS
+* `rbd`: bugs or enhancements related to RBD
+* `doc`: documentation updates
+* `util`: utilities shared between components use `cephfs` or `rbd` if the
+   change is only relevant for one of the type of storage
+* `journal`: any of the journalling functionalities
+* `helm`: deployment changes for the Helm charts
+* `build`: anything related to building Ceph-CSI, the executable or container
```
Comment (Madhu-1) [inline docs/development-guide.md:141]: let's add deploy or template to this one, it's good to go

Comment (nixpanic) [reply docs/development-guide.md:142]: Added `deploy` with comment about the Kubernetes templates.

File: docs/development-guide.md [hunk 7]
```diff
@@ -122,13 +122,31 @@ Signed-off-by: Random J Developer <random@developer.example.org>
...
+<component>: <subject of the change>
 <BLANK LINE>
-<why this change was made>
+<paragraph(s) with reason/description>
 <BLANK LINE>
-<footer>
+<signed-off-by>
 ```
 
+The `component` in the subject of the commit message can be one of the following:
```
Comment (humblec) [inline docs/development-guide.md:132]: @nixpanic its also good to add a tag called, dependency or vendor.

Comment (nixpanic) [reply docs/development-guide.md:132]: This is caught by `rebase` as dependencies or vendor updates normally update versions. New dependencies can also be introduced with `build` if it is a new direct dependency of ceph-csi.

Would that cover things sufficiently?

Comment (humblec) [reply docs/development-guide.md:132]: Yeah, indirectly `rebase` can include it. To be explicit, that tag helps.. but I am fine with either way. Let me know your choice, based on that we can go ahead and approve.

Comment (nixpanic) [reply docs/development-guide.md:132]: I prefer to stick with `rebase` for now. If we later on notice that `vendor` is something we would like to add, we can always do so.

**Pull request merged**